autovalet_goal_generation/src/autovalet_goal_generation/utils.py

```diff
@@ -0,0 +1,96 @@
+'''
+Contains utilities for goal generation. These helpers are reusable!
+
+Author  : subramak@andrew.cmu.edu
+Date    : 19 Oct 2020
+
+Changelog:
+    subbu - 10/19 - initial commit
+'''
+
+# python modules
+import numpy as np
+
+# ROS methods
+import rospy
+from tf.transformations import euler_from_quaternion, quaternion_from_euler
+from geometry_msgs.msg import PoseStamped
+
+def force2DGoal(pose):
+        '''
+        Input:
+        pose            : [PoseStamped] contains goal on which 2D constraints need to be enforced.
+        
+        Output:
+        goal_pose_2D    : [PoseStamped] contains the output goal pose which has z=0 and pitch, roll = 0
+        '''
+
+        goal_pose_2D = PoseStamped()
+
+        # repopulate unchanged contents for goal_pose_2D
+        # [doing it this way because I'm not sure if there's a copy constructor for PoseStamped messages]
+        goal_pose_2D.header = pose.header
+        goal_pose_2D.pose.position.x = pose.pose.position.x
+        goal_pose_2D.pose.position.y = pose.pose.position.y
+
+        # Perform corrections to make 2D nav goal
+        goal_pose_2D.pose.position.z = 0        # Force pose to ground
+        
+        # Force pitch and roll to zero
+        _, _, yaw = euler_from_quaternion((pose.pose.orientation.x,
+                                                  pose.pose.orientation.y,
+                                                  pose.pose.orientation.z,
+                                                  pose.pose.orientation.w))
+        x, y, z, w = quaternion_from_euler(0, 0, yaw)
+
+        goal_pose_2D.pose.orientation.x = x
+        goal_pose_2D.pose.orientation.y = y
+        goal_pose_2D.pose.orientation.z = z
+        goal_pose_2D.pose.orientation.w = w
+
+        return goal_pose_2D
```

Comment:
I'm using this in `finding_parking_spot.py` as well. Do you think we could share the code to avoid duplication? We can refactor later, after this MR. lmk.